### PR TITLE
Feature/cl/grid

### PR DESCRIPTION
**Pull request by hhdtan**
a somewhat functional grid

### CODE REVIEW

**Comment by hhdtan** [comment]
image

protesting my task by giving myself another task

File: OnLand/ControlLaptop/angular-UI/src/app/components/schematics/schematics.component.ts [hunk 1]
```diff
@@ -113,8 +127,176 @@ export class SchematicsComponent implements OnInit {
...
+                // note: I have no idea how to rotate the text 90d ccw, flip is not the answer 
+                //flip: go.GraphObject.FlipHorizontal,
+                segmentOrientation: go.Link.OrientOpposite,
+                interval: 5,
+                alignmentFocus: go.Spot.BottomLeft,
+                segmentOffset: new go.Point(0, -20)
+                })
+            );
+
+        // var posiIndicator =
```
Comment (yeeteing) [inline OnLand/ControlLaptop/angular-UI/src/app/components/schematics/schematics.component.ts:179]: can we remove the comment?

File: OnLand/ControlLaptop/angular-UI/src/app/components/schematics/schematics.component.ts [hunk 2]
```diff
@@ -113,8 +127,176 @@ export class SchematicsComponent implements OnInit {
...
+            { gridCellSize: new go.Size(10, 10) }, // width x height of the grid system
+
+            // XY lines at specified interval
+            $(go.Shape, "LineH", { stroke: "grey", interval: 10 }), 
+            $(go.Shape, "LineH", { stroke: "lightgrey", interval: 20 }), 
+            $(go.Shape, "LineV", { stroke: "grey", interval: 10 }),
+            $(go.Shape, "LineV", { stroke: "lightgrey", interval: 20 }),
+
+            // XY lines for entire grid 
+            // $(go.Shape, "LineH", { stroke: "pink" }),
```
Comment (yeeteing) [inline OnLand/ControlLaptop/angular-UI/src/app/components/schematics/schematics.component.ts:141]: can we remove the comment?

File: OnLand/ControlLaptop/angular-UI/src/app/components/schematics/schematics.component.ts [hunk 3]
```diff
@@ -113,8 +127,176 @@ export class SchematicsComponent implements OnInit {
 				$(go.Shape,
 					new go.Binding("stroke", "color"),  // shape.stroke = data.color
 					new go.Binding("strokeWidth", "thick")));
+        
+        
+        // ---GRID START---  
+        dia.grid.visible = true; 
+        dia.toolManager.draggingTool.isGridSnapEnabled = true; // snappy snap
+        // dia.toolManager.resizingTool.isGridSnapEnabled = true;
```
Comment (yeeteing) [inline OnLand/ControlLaptop/angular-UI/src/app/components/schematics/schematics.component.ts:129]: do we need this ?

**Review by yeeteing — COMMENTED** [review]
Look good to me, added a few comments.

**Pull request merged**